docker/package/packages.py

```diff
@@ -272,11 +279,13 @@ def mk_node_unit(
             systemd_units=node_units,
             postinst_steps=node_postinst_steps,
             postrm_steps=node_postrm_steps,
+            supply_additional_files=node_supply_additional_files,
```

Comment:
I'm not sure that I understand the addition of this `supply_additional_files` field.

AFAICT its only definition is `node_supply_additional_files` and it is only set for nodes, so it doesn't seem great to add it (empty/`None`) to all packages only for these ones.

TBC: since this avoids duplication IMO it should remain a function that can be used for both ubuntu and fedora, just not one to be included in the `AbstractPackage` class

Reply:
For now it's relevant only tor `tezos-node`, yes, but it could be useful in the future for another package.
There is actually problem, that `supply_additional_files` function make sense when and only when `additional_files` field is not empty. 
Instead of that two fields I propose to have one field with list of pairs: additional file name and function for delivering it.
Thus, if that list is empty, it just means that package does not have any additional files to install except for main binary and services-related scripts.

Reply:
> it could be useful in the future for another package.

:thinking: fair, but the question is: would `supply_additional_files` be any different from `node_supply_additional_files` for them?

I may be wrong, but I think that a single function might do the job. IOW what I'm suggesting is to keep only `node_supply_additional_files`  (renamed) without adding the `supply_additional_files` field.

In that case `additional_files` would be the only field needed (and of course nothing should happen if this is empty).

Reply:
At first I did it that way, but the only place it could happen is `ubuntu.py` or `fedora.py`, where generic package builds, it doesn't know it's name beforehand
That's why i put it in the package definition, it would be quite bad instructions for `tezos-node` specifically in the generic package build script

Reply:
> it doesn't know it's name beforehand

sorry, I didn't get that, what do you mean by that?

> it would be quite bad to put instructions specifically for tezos-node in the generic package build script

that I agree, if the installation process is specific to `tezos-node` it shouldn't be added to every other package.

However I don't understand why it is :thinking: I would prefer this to be as simple as possible.

Doing this in the post-install phase allowed us to keep a common main phase and put this node-specific stuff there. The problem is that these files are unknown to packages, but that could be changed AFAIK.

If we modify this approach (and we need to keep it node-specific) we need to pick one that's simpler and more robust and this one doesn't look like that to me TBH :shrug:

Reply:
So, I have a few questions:

First: if we keep the post-install approach and add a list for these files, would that be enough? 

Second: why do you think that the approach currently in this PR is better than keeping things in post-install? 

Lastly, if we change the main phase, isn't all that needs to be added:
1. files copied from a local copy
2. symlinks added from said files

then isn't it enough to add one (or two at most) list of strings for which we do that (for every package)?

I'm asking this last question mostly to understand, but I also don't think that this is necessarily better than keeping the post-install approach.

Reply:
> that I agree, if the installation process is specific to tezos-node it shouldn't be added to every other package.

Reply:
> First: if we keep the post-install approach and add a list for these files, would that be enough?

I believe keeping post-install approach is semantically incorrect and that's why it already caused some trouble

Reply:
> Lastly, if we change the main phase, isn't all that needs to be added:

Reply:
The pros of current approach is simplicity, the cons is less flexibility (we can only install additional scripts now and nothing more, but I'm not sure if we will ever need something more complicated)

Reply:
> What do you think about it?

I'd prefer to indeed remove the `supply_additional_files` and instead only adding a list of scripts to install as part of the main phase.

I'd also like not to keep the current names, as they are more convenient for users (and also to avoid a breaking change).

Now, as you correctly pointed out, a list with only the script names isn't sufficient, because of the renaming, so I think that the best approach would be a list of objects. Each object should contain:
1. the local name of the script
2. (optionally, only if different) the installed name of the script
3. (optionally, only if desired) the name for a symlink to the installed script

This is simpler than the current approach in this PR IMO (at least to use in the future) and should have enough flexibility as well.

Of course, the main installation phase needs to be modified so that (instead of running the `supply_additional_files` lambda) for each of the `additional_files` objects:
1. the script is installed with the given name (or the same one, if no renaming is necessary)
2. a symlink is created for it (if one is required)

Does that sound doable to you?

Reply:
There is updated implementation now, don't you like it? I'll implement as you offered if so

Reply:
But it's still not that simple. Not only script's name changes, but the contents too: `octez-node-kathmandunet` contains reference to `tezos-node-kathmandunet.service` and etc.

Reply:
> There is updated implementation now

Yes, I've seen that already, but it's still based on the `additional_files_install`, so the point still applies.

> Not only script's name changes, but the contents too

And that should remain the case, no? I mean, we don't want to rename `tezos-node-kathmandunet.service` for example.

I don't get why this is a problem, can you please clarify?